src/components/PreferencesModal/KeymapTab.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { useCallback, useMemo } from 'react'
+import { Section, SectionHeader } from './styled'
+import { usePreferences } from '../../lib/preferences'
+import {
+  getGenericShortcutString,
+  KeymapItem,
+  KeymapItemEditableProps,
+} from '../../lib/keymap'
+import { useTranslation } from 'react-i18next'
+import KeymapItemSection from '../atoms/KeymapItemSection'
+import styled from '../../lib/styled/styled'
+
+const KeymapTab = () => {
+  const {
+    preferences,
+    updateKeymap,
+    removeKeymap,
+    resetKeymap,
+  } = usePreferences()
+  const { t } = useTranslation()
+
+  const keymap = useMemo(() => {
+    const keymap = preferences['general.keymap']
+    if (keymap != null) {
+      return [...keymap.entries()]
+    } else {
+      return []
+    }
+  }, [preferences])
+
+  const handleAssignNewKeymap = useCallback(
+    (
+      key: string,
+      firstShortcut: KeymapItemEditableProps,
+      secondShortcut?: KeymapItemEditableProps
+    ) => {
+      return updateKeymap(key, firstShortcut, secondShortcut)
+    },
+    [updateKeymap]
+  )
+
+  const handleRemoveKeymap = useCallback(
+    (key: string) => {
+      removeKeymap(key)
+    },
+    [removeKeymap]
+  )
+
+  const handleResetKeymap = useCallback(() => {
+    resetKeymap()
+  }, [resetKeymap])
+
+  const getKeymapItemSectionKey = useCallback((keymapItem: KeymapItem) => {
+    if (keymapItem.shortcutMainStroke == null) {
+      return keymapItem.description
+    } else {
+      return getGenericShortcutString(keymapItem.shortcutMainStroke)
+    }
+  }, [])
+
+  return (
+    <Section>
+      <KeymapHeaderSection>
+        <SectionHeader>{t('preferences.keymap')}</SectionHeader>
+        <SectionResetKeymap>
+          <ButtonContainer onClick={handleResetKeymap}>Reset</ButtonContainer>
+        </SectionResetKeymap>
+      </KeymapHeaderSection>
+      <KeymapItemList>
+        {keymap != null &&
+          keymap.map((keymapEntry: [string, KeymapItem]) => {
+            return (
+              <KeymapItemSection
+                key={getKeymapItemSectionKey(keymapEntry[1])}
+                keymapKey={keymapEntry[0]}
+                currentKeymapItem={keymapEntry[1].shortcutMainStroke}
+                description={keymapEntry[1].description}
+                onAssignNewKeymap={handleAssignNewKeymap}
```

Comment:
I'd like to use a simpler name `updateKeymap`

Reply:
Agree